samza-azure/src/main/java/org/apache/samza/scheduler/HeartbeatScheduler.java

```
@@ -0,0 +1,79 @@
+ * under the License.
+ */
+
+package org.apache.samza.scheduler;
+
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+import org.apache.samza.TableUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Scheduler invoked by each processor for heartbeating to a row of the table.
+ * Heartbeats every 5 seconds.
+ * The row is determined by the job model version and processor id passed to the scheduler.
+ * All time units are in SECONDS.
+ */
+public class HeartbeatScheduler implements TaskScheduler {
+
+  private static final Logger LOG = LoggerFactory.getLogger(HeartbeatScheduler.class);
+  private static final long HEARTBEAT_DELAY_SEC = 5;
+  private static final ThreadFactory PROCESSOR_THREAD_FACTORY =
+      new ThreadFactoryBuilder().setNameFormat("HeartbeatScheduler-%d").build();
+  private final ScheduledExecutorService scheduler = Executors.newScheduledThreadPool(5, PROCESSOR_THREAD_FACTORY);
+  private final String processorId;
+  private final TableUtils table;
+  private final AtomicReference<String> currentJMVersion;
+  private SchedulerStateChangeListener listener = null;
+
+  public HeartbeatScheduler(TableUtils table, AtomicReference<String> currentJMVersion, final String pid) {
+    this.table = table;
+    this.currentJMVersion = currentJMVersion;
+    processorId = pid;
+  }
+
+  @Override
+  public ScheduledFuture scheduleTask() {
+    return scheduler.scheduleWithFixedDelay(() -> {
+        try {
+          LOG.info("Updating heartbeat");
+          table.updateHeartbeat(currentJMVersion.get(), processorId);
+        } catch (Exception e) {
+          LOG.error("Exception in Heartbeat Scheduler.", e);
```

It is perhaps useful to provide a handler for exceptions. Otherwise, we are just swallowing the exceptions. 
Option 1: change `schedulerTask` interface to take in a lambda for error handling
Option 2: if every submitted task to this scheduler is going to use the same error handler, simply pass lambda/handler in the constructor